src/line_opacity.jl

```diff
@@ -0,0 +1,147 @@
+"""
+    line_opacity(linelist, wls, temp, n_density, abundances; window_size)
+
+Calculate the opacity coefficient, α, from all lines in `linelist`, at wavelengths `wls`.
+
+other arguments:
+- `temp` the temerature in K
+- `total_n_density` the number density of non-electron particles in cm^-3
+- `abundances` should be a `Dict` mapping species to absolute abundances between 0 and 1.
+- `window_size` (optional, default: 20), the maximum distance from the line center at which line 
+opacities should be calculated in included.
+"""
+function line_opacity(linelist, wls, temp, electron_density, total_n_density, abundances::Dict, 
+                      atomic_masses::Dict, partition_fns::Dict, ionization_energies::Dict
+                      ; window_size=20.0)
+    #I don't love passing all these dictionaries arround, but I also don't really want to access 
+    #them as globals.  What do you think?  Maybe globals for some of them wouldn't be so bad?
```

Comment:
In general, I think it's better to pass things around explicitly (otherwise, I think the code's control flow and dependencies become a lot harder to reason about - especially if you've never seen it before). 

Maybe you could use globals to define default arguments? Or maybe you could package abundances & abundances & partition functions & ionization energies into a separate data structure?

Reply:
What do you think about turning parition_funcs and ionization_energies into functions rather that dicts. I think it's reasonable to expect that users won't want to replace them, in which case they "want" to be global, but module-level functions don't have the same confusing state problems that module level variables do.

That's just spitballing, I'm still brainstorming while I try to get a minimal example working.

Reply:
That could potentially work. I don't really see any problems with making them functions.

I do think it's important that we make it easy to replace these things for 2 reasons:
1. Our partition functions only include data for Temperatures up to 10,000 K and they don't have cutoff energy levels. Therefore, we will need different partition function data for hot stars
2. When comparing to MOOG or Turbospectrum, we may want to directly reuse their partition functions/ionization energies for the most fair comparison.

I'm a little concerned that hardcoding in a single function could make debugging issues in either of the above 2 cases much more difficult. I think I still have a slight preference for explicitly passing in this data (in some kind of atomic/molecular data struct). At the same time, I recognize that I have a tendency to future proof and this may not be necessary. I'm happy to defer to your preference on this.

Reply:
I think I've got a potential compromise. What if you make the partition function/ionization energy function configurable at the global module level?

I'm finding it hard to describe exactly what I'm thinking, so I'll use an example. Imagine that we had 2 or 3 different partition function implementations. For the sake of example, let's refer to the implementations as
- ``bc16_partition_func``
- ``moog_partition_func``
- ``turbospec_partition_func``
The idea would be to define some function at the top level of the module called something like ``sssynth_partition_func`` that would wrap one of implementations. 

I don't love it, but this might address my concerns about switching between implementations and your concerns about passing around too many arguments. It's just an idea. Feel free to do whatever you think is best (preferably in a separate PR)

Reply:
I accidentally replied to the main thread. See below.